File: Try_GameEngine5/src/com/example/try_gameengine/framework/TouchHandler.java
package com.example.try_gameengine.framework;

public abstract class TouchHandler {

	protected ITouchable delegate;
	protected int priority;
	protected boolean claimed;
	protected boolean consumeTouch;

	public TouchHandler(ITouchable touch, int priority){
		this(touch, priority, true);
	}
	
	public TouchHandler(ITouchable touch, int priority, boolean consumeTouch){
		this.delegate = touch;
		this.priority = priority;
		this.consumeTouch = consumeTouch;
	}

	public ITouchable getDelegate() {
		return delegate;
	}

	public void setDelegate(ITouchable delegate) {
		this.delegate = delegate;
	}

	public int getPriority() {
		return priority;
	}

	public void setPriority(int priority) {
		this.priority = priority;
	}

	public boolean isClaimed() {
		return claimed;
	}

	public void setClaimed(boolean claimed) {
		this.claimed = claimed;
	}

	public boolean isConsumeTouch() {
		return consumeTouch;
	}

	public void setConsumeTouch(boolean consumeTouch) {
		this.consumeTouch = consumeTouch;
	}

}

File: Try_GameEngine5/src/com/example/try_gameengine/framework/TouchHandler.kt
package com.example.try_gameengine.framework

abstract class TouchHandler @JvmOverloads constructor(
    touch: ITouchable?,
    priority: Int,
    consumeTouch: Boolean = true
) {
    @JvmField
    var delegate: ITouchable?
    @JvmField
    var priority: Int
    var isClaimed: Boolean = false
    @JvmField
    var isConsumeTouch: Boolean

    init {
        this.delegate = touch
        this.priority = priority
        this.isConsumeTouch = consumeTouch
    }

    @get:kotlin.jvm.JvmName("getClaimedProperty")
    @set:kotlin.jvm.JvmName("setClaimedProperty")
    var claimed: Boolean
        get() = isClaimed
        set(value) {
            isClaimed = value
        }

    fun getDelegate(): ITouchable? = delegate

    fun getPriority(): Int = priority

    fun isConsumeTouch(): Boolean = isConsumeTouch
}
